Sources/PackageCollectionsSigning/PackageCollectionSigning.swift

```diff
@@ -8,23 +8,65 @@
  See http://swift.org/CONTRIBUTORS.txt for Swift project authors
  */
 
+import Dispatch
 import Foundation
 
+import Basics
 import PackageCollectionsModel
+import TSCBasic
 
 public struct PackageCollectionSigning {
     public typealias Model = PackageCollectionModel.V1
 
     private static let minimumRSAKeySizeInBits = 2048
 
-    let certPolicy: CertificatePolicy
+    /// URL of the optional directory containing root certificates to be trusted.
+    private let trustedRootCertsDir: URL?
+
+    /// The `DispatchQueue` to use for callbacks
+    private let callbackQueue: DispatchQueue
+    /// Diagnostics engine to emit warnings and errors
+    private let diagnosticsEngine: DiagnosticsEngine?
+
+    /// Internal cache/storage of `CertificatePolicy`s
+    private let certPolicies = ThreadSafeKeyValueStore<CertificatePolicyKey, CertificatePolicy>()
+
+    public init(trustedRootCertsDir: URL? = nil, callbackQueue: DispatchQueue = DispatchQueue.global(), diagnosticsEngine: DiagnosticsEngine? = nil) {
```

Comment:
nit: I would make the diagnostics engine non-optional, its easy enough to instantiate one in tests